gridfs/grid_file.py

```diff
@@ -656,33 +658,40 @@ def seekable(self):
     def __iter__(self):
         """Return an iterator over all of this file's data.
 
-        The iterator will return chunk-sized instances of
-        :class:`str` (:class:`bytes` in python 3). This can be
-        useful when serving files using a webserver that handles
-        such an iterator efficiently.
-
-        .. note::
-           This is different from :py:class:`io.IOBase` which iterates over
-           *lines* in the file. Use :meth:`GridOut.readline` to read line by
-           line instead of chunk by chunk.
+        The iterator will return lines (delimited by b'\n') of
+        :class:`bytes`. This can be useful when serving files
+        using a webserver that handles such an iterator efficiently.
 
         .. versionchanged:: 3.8
            The iterator now raises :class:`CorruptGridFile` when encountering
            any truncated, missing, or extra chunk in a file. The previous
            behavior was to only raise :class:`CorruptGridFile` on a missing
            chunk.
+
+        .. versionchanged:: 4.0
+            The iterator now iterates over *lines* in the file, instead
+            of chunks, to conform to the base class :py:class:`io.IOBase`.
+            Use :meth:`GridOut.readchunk` to read chunk by chunk instead
+            of line by line.
         """
-        return GridOutIterator(self, self.__chunks, self._session)
+        return self
```

Comment:
The existing IOBase iterator implementation already does all the work for the iterator by using `readline` in `__next__`, so `GridOutIterator` is not needed anymore as far as I can tell, IOBase already does all the work.